core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -269,32 +275,86 @@ else if(listener.getAuthenticationMethod().equals("HEADER")) {
 				messageContext.put(paramname, paramvalue);
 			}
 
+			/**
+			 * Map headers into messageContext
+			 */
+			Enumeration<String> headers = request.getHeaderNames();
+			XmlBuilder headersXml = new XmlBuilder("headers");
+			while (headers.hasMoreElements()) {
+				String header = headers.nextElement().toLowerCase();
+				if(IGNORE_HEADERS.contains(header))
```

Comment:
Het lijkt me verstandig om niet zomaar alle headers te kopieren, maar om in een whitelist mechanisme te voorzien.